packages/data-context/src/actions/ProjectActions.ts

```diff
@@ -86,10 +86,10 @@ export class ProjectActions {
     this.ctx.update((d) => {
       d.activeBrowser = null
       d.currentProject = null
-      d.diagnostics = {
-        error: null,
-        warnings: [],
-      }
+
+      d.diagnostics.error = null
+      d.diagnostics.warnings = []
+      d.diagnostics.globallyIssuedWarnings = new Set()
```

Comment:
We bust this warning cache when we navigate out of the project (eg go to global mode) same as prior to this PR. I just moved the source of truth from `config` to `data-context`.